cbcontainers/state/components/sensor_daemon_set.go

```diff
@@ -536,7 +538,8 @@ func (obj *SensorDaemonSetK8sObject) mutateClusterScannerEnvVars(container *core
 		WithEnvVarFromResource("CLUSTER_SCANNER_REQUESTS_MEMORY", ClusterScanningContainerName, "requests.memory").
 		WithEnvVarFromField("CLUSTER_SCANNER_NODE_NAME", "spec.nodeName", "v1").
 		WithSpec(clusterScannerSpec.Env).
-		WithGatewayTLS()
+		WithGatewayTLS().
+		WithProxySettings(agentSpec.Components.Settings.Proxy)
```

Comment:
For the **cluster-scanner**, we want to unpack the `cbcontainers-image-scanning-reporter.cbcontainers-dataplane.svc.cluster.local` NO_PROXY value by default (unless it was already set by the customer). 

That is aim to simplify the customer CRD setting - customers was always forgetting to fill this value (documented [here](https://github.com/octarinesec/octarine-operator#using-http-proxy) ) or filled it incorrectly. That is why we wanted to fill them ourselves in the operator.